commands/deploy.go

```diff
@@ -343,9 +343,48 @@ func (p *deploycmd) deployFuncV20180708(c *cli.Context, app *models.App, funcfil
 	if funcfile.Name == "" {
 		funcfile.Name = filepath.Base(filepath.Dir(funcfilePath)) // todo: should probably make a copy of ff before changing it
 	}
-	fmt.Printf("Deploying %s to app: %s\n", funcfile.Name, app.Name)
 
-	var err error
+	oracleProvider, _ := getOracleProvider()
+	if oracleProvider != nil && oracleProvider.ImageCompartmentID != "" {
+		// If the provider is Oracle and ImageCompartmentID is present, we need to deploy image to the ImageCompartmentID.
+
+		// Check Registry and build the repository name.
+		if client.Registry() == "" {
+			return errors.New("registry name must be set, be sure to configure your context file")
+		}
+		res, err := getRepositoryNamePrefix(client.Registry())
+		if err != nil {
+			return err
+		}
+		repositoryName := res + "/" + funcfile.Name
+
+		// Create artifactsClient and set region.
+		artifactsClient, err := artifacts.NewArtifactsClientWithConfigurationProvider(oracleProvider.ConfigurationProvider)
+		if err != nil {
+			return err
+		}
+		artifactsClient.SetRegion(getRegion(oracleProvider))
+
+		// Create the repository in ImageCompartmentID
+		repositoryExists, err := createContainerRepositoryInCompartment(repositoryName, oracleProvider.ImageCompartmentID, artifactsClient)
```

Comment:
I wonder if the logic in this area might be more clear if it was switched around slightly?

What do you think about checking for existence of the container first, with `doesRepositoryExistInComparment`. If that's true, then you're done. If false, then proceed to create the repository, and if that fails, you know it's a failure that you can't continue with.

That way the response from `createContainerRepositoryInCompartment` doesn't have to distinguish between 409s and other responses.

Reply:
Nice catch. Updating it that way